ui/app/src/state/epics/contentTypes.ts

```diff
@@ -65,9 +65,15 @@ export default [
       ofType(fetchComponentsByContentType.type, setContentTypeFilter.type),
       withLatestFrom(state$),
       switchMap(([, state]) => {
-        const allowedContentTypes = Object.entries(state.preview.guest?.allowedContentTypes ?? {}).flatMap(
+        let allowedContentTypes = Object.entries(state.preview.guest?.allowedContentTypes ?? {}).flatMap(
           ([key, type]) => (type.shared ? [key] : [])
         );
+        // If '*' is included in allowedContentTypes, it means that all content types are allowed.
+        if (allowedContentTypes.includes('*')) {
+          allowedContentTypes = Object.values(state.contentTypes.byId)
+            .filter((contentType) => contentType.type === 'component' && !contentType.id.includes('/level-descriptor'))
+            .map((contentType) => contentType.id);
+        }
```

Comment:
Above, you can see the flatMap checks for `.shared`. I'm not sure that's considered here. Please double check.

Reply:
There was an error, since the property we use for allowing existing components is 'sharedExisting'. So I updated it in here and in PreviewBrowseComponentsPanel.

This epic here handles two actions:
- fetchComponentsByContentType: Used only by PreviewBrowseComponentsPanel. This one has a scenario where we fetch only compatible components. For this one we define `allowedContentTypes`.
- setContentTypeFilter: Used by PreviewBrowseComponents, PreviewComponentsPanel and PreviewInPageInstancePanel. For this action `allowedContentTypes` is not used since the fetch is done against a specific type, and we fetch components no matter if they're not compatible.

So, the update in `allowedContentTypes` won't affect usages outside PreviewBrowseComponentsPanel.